pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go

```diff
@@ -271,11 +290,15 @@ func (m *Manager) processShutdownEvent() error {
 	case <-c:
 		break
 	case <-time.After(m.shutdownGracePeriodRequested):
-		klog.V(1).InfoS("Shutdown manager pod killing time out", "gracePeriod", m.shutdownGracePeriodRequested)
+		msg := "Shutdown manager pod killing time out"
+		m.recorder.Event(m.nodeRef, v1.EventTypeWarning, kubeletevents.NodeShutdown, msg)
+		klog.V(1).InfoS(msg, "gracePeriod", m.shutdownGracePeriodRequested)
 	}
 
 	m.dbusCon.ReleaseInhibitLock(m.inhibitLock)
-	klog.V(1).InfoS("Shutdown manager completed processing shutdown event, node will shutdown shortly")
+	msg := "Shutdown manager completed processing shutdown event, node will shutdown shortly"
+	m.recorder.Event(m.nodeRef, v1.EventTypeNormal, kubeletevents.NodeShutdown, msg)
```

Comment:
I'm not sure here, but it's unclear if this event will actually be sent. As soon as the inhibitor lock is released (on line 298) systemd will shutdown the machine. So it's unclear if this code will even run. In the case it does, I'm not sure there will be enough time for the event to actually get sent...

Reply:
fixed